### PR TITLE
Add OmnipodDashPodStateManager

### PR DESCRIPTION
**Pull request by bartsopers**
If you could merge `dash` into your branch first, the diff will be a lot smaller.

### CODE REVIEW

**Review by AdrianLxM — COMMENTED** [review]
a few remarks 🗡️

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/comm/io/BleIO.kt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    fun receivePacket(characteristic: CharacteristicType): ByteArray {
+        synchronized(state) {
+            if (state != IOState.IDLE) {
+                throw BleIOBusyException()
+            }
+            state = IOState.READING
+        }
+        val ret = incomingPackets[characteristic]?.poll(DEFAULT_IO_TIMEOUT_MS.toLong(), TimeUnit.MILLISECONDS)
+            ?: throw TimeoutException()
+        synchronized(state) { state = IOState.IDLE }
```
[the remark this answers is not in the document]
Comment (bartsopers) [reply omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/comm/io/BleIO.kt:34]: This is @avereha's code, it has already been merged to `dash`

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/dagger/OmnipodDashModule.kt [hunk 2]
```diff
@@ -31,6 +37,15 @@ abstract class OmnipodDashModule {
     @ContributesAndroidInjector
     abstract fun contributesOmnipodDashOverviewFragment(): OmnipodDashOverviewFragment
 
-    @ContributesAndroidInjector
-    abstract fun contributesBleManager(): BleManager
-}
+    companion object {
+
+        @Provides
+        fun providesBleManager(bleManager: BleManager): OmnipodDashBleManager = bleManager
```
[the remark this answers is not in the document]
Comment (bartsopers) [reply omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/dagger/OmnipodDashModule.kt]: Fixed.
Thank you for the explanation :)

**Review by AdrianLxM — APPROVED** [review]
lgtm 👍

File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/comm/io/BleIO.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (avereha) [reply omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/comm/io/BleIO.kt:34]: I am still working on this part. 
In order to test edge cases and failure modes, I have to implement more(packet splitting and flow control) first.

My initial intention is to set the state back to IDLE in the not implemented yet `flushIncomingQueues` function that gets called before sending a message(that is a collection of small packets&flow control): https://github.com/0pen-dash/AndroidAPS/pull/4/files#diff-99d07a365576378831736fd6dfedae548c3c55791e00d1cd50188279af9a3699R17
